server/sandbox_run_linux.go

```diff
@@ -587,15 +587,17 @@ func (s *Server) runPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
 	sb.SetInfraContainer(container)
 
 	var ip string
-	var result cnitypes.Result
+	var results []cnitypes.Result
 
 	if s.config.Config.ManageNetworkNSLifecycle {
-		ip, result, err = s.networkStart(sb)
+		ip, results, err = s.networkStart(sb)
 		if err != nil {
 			return nil, err
 		}
-		if result != nil {
-			g.AddAnnotation(annotations.CNIResult, result.String())
+		if len(results) > 0 {
+			for _, result := range results {
+				g.AddAnnotation(annotations.CNIResult, result.String())
```

Comment:
This is overwriting the value with the last one being the winner.

Reply:
Should we, rather than getting all results, just choose the first, and base everything off of it? I can't think of a better way of deciding which config is the correct one to use, both for the annotation and ip